app/code/Magento/Catalog/Model/Product/Url.php

```diff
@@ -19,46 +25,53 @@ class Url extends \Magento\Framework\DataObject
     /**
      * URL instance
      *
-     * @var \Magento\Framework\UrlFactory
+     * @var UrlFactory
      */
     protected $urlFactory;
 
     /**
-     * @var \Magento\Framework\Filter\FilterManager
+     * @var FilterManager
      */
     protected $filter;
 
     /**
      * Store manager
      *
-     * @var \Magento\Store\Model\StoreManagerInterface
+     * @var StoreManagerInterface
      */
     protected $storeManager;
 
     /**
-     * @var \Magento\Framework\Session\SidResolverInterface
+     * @var SidResolverInterface
      */
     protected $sidResolver;
 
     /**
-     * @var \Magento\UrlRewrite\Model\UrlFinderInterface
+     * @var UrlFinderInterface
      */
     protected $urlFinder;
 
     /**
-     * @param \Magento\Framework\UrlFactory $urlFactory
-     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
-     * @param \Magento\Framework\Filter\FilterManager $filter
-     * @param \Magento\Framework\Session\SidResolverInterface $sidResolver
-     * @param UrlFinderInterface $urlFinder
-     * @param array $data
+     * @var ScopeConfigInterface
+     */
+    protected $scopeConfig;
+
+    /**
+     * @param UrlFactory            $urlFactory
+     * @param StoreManagerInterface $storeManager
+     * @param FilterManager         $filter
+     * @param SidResolverInterface  $sidResolver
+     * @param UrlFinderInterface    $urlFinder
+     * @param ScopeConfigInterface  $scopeConfig
+     * @param array                 $data
      */
     public function __construct(
-        \Magento\Framework\UrlFactory $urlFactory,
-        \Magento\Store\Model\StoreManagerInterface $storeManager,
-        \Magento\Framework\Filter\FilterManager $filter,
-        \Magento\Framework\Session\SidResolverInterface $sidResolver,
+        UrlFactory $urlFactory,
+        StoreManagerInterface $storeManager,
+        FilterManager $filter,
+        SidResolverInterface $sidResolver,
         UrlFinderInterface $urlFinder,
+        ScopeConfigInterface $scopeConfig,
```

Comment:
Can you please add this as an optional dependancy and add it after the `array $data = []`. By making this change we will be able to keep your code as a backwards compatible change. See https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/index.html#adding-a-constructor-parameter for more information.